traitlets/traitlets.py

```diff
@@ -943,8 +943,8 @@ def info(self):
         if isinstance(self.klass, py3compat.string_types):
             klass = self.klass
         else:
-            klass = self.klass.__name__
-        result = 'a subclass of ' + klass
+            klass = self.klass.__module__+'.'+self.klass.__name__
+        result = "a subclass of '" + klass + "'"
```

Comment:
I think you've been writing too much javascript. In Python, we have string formatting, so you can use:

```python
"a subclass of '%s'" % klass
```